File: Section1 - Basics/Math/problem2.js
// Brute Force
function reversedBits(N) {
	let res = (N >>> 0).toString(2);
	let parity = 32 - res.length;
	if (parity > 0) {
		res = "0".repeat(parity) + res;
	}
	res = res.split("").reverse().join("");
	let decimal = parseInt(res, 2);
	console.log(decimal);
}

reversedBits(12);

File: Section1 - Basics/Math/problem2.java
// Brute Force
Long reversedBits(Long X) {
    // code here
    String res = Long.toBinaryString(X);
    int parity = 32 - res.length();
    if(parity>0) {
        res = "0".repeat(parity) + res;
    }
    StringBuilder revRes = new StringBuilder();
    revRes.append(res);
    revRes.reverse();
    String revStr = revRes.toString();
    long decimal=Long.parseLong(revStr,2);  
    return decimal;
  
}

